connector/connect/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectCommandPlannerSuite.scala

```diff
@@ -0,0 +1,172 @@
+   * @param thunk
+   */
+  def withTable(thunk: String => Any): Unit = {
+    val name = table()
+    thunk(name)
+    spark.sql(s"drop table if exists ${name}")
+  }
+
+  /**
+   * Helper method that takes a closure as an arugment and handles cleanup of the file system
+   * resource created.
+   * @param thunk
+   */
+  def withPath(thunk: String => Any): Unit = {
+    val name = path()
+    thunk(name)
+    FileUtils.deleteDirectory(new File(name))
+  }
+
+  def transform(cmd: proto.Command): Unit = {
+    new SparkConnectCommandPlanner(spark, cmd).process()
+  }
+
+  test("Writes fails without path or table") {
+    assertThrows[UnsupportedOperationException] {
+      transform(localRelation.write())
+    }
+  }
+
+  test("Write fails with unknown table - AnalysisException") {
+    val cmd = readRel.write(tableName = Some("dest"))
+    assertThrows[AnalysisException] {
+      transform(cmd)
+    }
+  }
+
+  test("Write with partitions") {
+    val name = table()
+    val cmd = localRelation.write(
+      tableName = Some(name),
+      format = Some("parquet"),
+      partitionByCols = Seq("noid"))
+    assertThrows[AnalysisException] {
+      transform(cmd)
+    }
+  }
+
+  test("Write with invalid bucketBy configuration") {
+    val cmd = localRelation.write(bucketByCols = Seq("id"), numBuckets = Some(0))
+    assertThrows[InvalidCommandInput] {
+      transform(cmd)
+    }
+  }
+
+  test("Write to Path") {
+    withPath { name =>
+      val cmd = localRelation.write(format = Some("parquet"), path = Some(name))
+      transform(cmd)
+      assert(Files.exists(Paths.get(name)), s"Output file must exist: ${name}")
+    }
+  }
+
+  test("Write to Path with invalid input") {
+    // Wrong data source.
+    assertThrows[SparkClassNotFoundException](
+      transform(localRelation.write(path = Some(path), format = Some("ThisAintNoFormat"))))
+
+    // Default data source not found.
+    assertThrows[SparkClassNotFoundException](transform(localRelation.write(path = Some(path))))
+  }
+
+  test("Write with sortBy") {
+    // Sort by existing column.
+    transform(
+      localRelation.write(
+        tableName = Some(table),
+        format = Some("parquet"),
+        sortByColumns = Seq("id"),
+        bucketByCols = Seq("id"),
+        numBuckets = Some(10)))
+
+    // Sort by non-existing column
+    assertThrows[AnalysisException](
+      transform(
+        localRelation
+          .write(
+            tableName = Some(table),
+            format = Some("parquet"),
+            sortByColumns = Seq("noid"),
+            bucketByCols = Seq("id"),
+            numBuckets = Some(10))))
+  }
+
+  test("Write to Table") {
+    withTable { name =>
+      val cmd = localRelation.write(format = Some("parquet"), tableName = Some(name))
+      transform(cmd)
+      // Check that we can find and drop the table.
+      spark.sql(s"select count(*) from ${name}").collect()
```

Comment:
Should we compare the results? Could leverage `checkAnswer` at `QueryTest`

Reply:
The table is empty and the assumption is that the write operation was successful if the table can be found, because the write is already implemented and tested.